components/team/MemberEditor.tsx

```diff
@@ -0,0 +1,209 @@
+  onSuccess,
+}: {
+  member: ToolsOzoneTeamDefs.Member | null
+  onCancel: () => void
+  onSuccess: () => void
+}) {
+  const { onFormSubmit, submission } = useMemberEditor({
+    isNewMember: !member,
+    onSuccess,
+  })
+  return (
+    <Card className="mb-3">
+      <form action="" onSubmit={onFormSubmit}>
+        <div className="mb-3">
+          <FormLabel label="Handle/DID" htmlFor="identifier" className="flex-1">
+            {member && <input value={member.did} hidden name="did" />}
+            <Input
+              required
+              type="text"
+              id="identifier"
+              name="identifier"
+              autoFocus={!member}
+              className="block w-full"
+              value={member?.did}
+              disabled={!!member || submission.isSubmitting}
+              placeholder="member.bsky.social or did:plc:...."
+            />
+          </FormLabel>
+        </div>
+        <div className="flex flex-row items-center mb-3 gap-4">
+          <div className="">
+            <FormLabel label="Role" htmlFor="role" className="flex-1">
+              <Select
+                required
+                id="role"
+                name="role"
+                autoFocus={!!member}
+                disabled={submission.isSubmitting}
+              >
+                {Object.entries(MemberRoleNames).map(([role, name]) => (
+                  <option
+                    key={role}
+                    value={role}
+                    selected={member?.role === role}
+                  >
+                    {name}
+                  </option>
+                ))}
+              </Select>
+            </FormLabel>
+          </div>
+          {!!member && (
+            <div>
+              <FormLabel
+                label="Access Control"
+                htmlFor="disabled"
+                className="flex-1"
+              >
+                <Checkbox
+                  value="true"
+                  id="disabled"
+                  name="disabled"
+                  className="mb-3 flex items-center"
+                  disabled={submission.isSubmitting}
+                  label="Disable access for this member"
+                />
+              </FormLabel>
+            </div>
+          )}
+        </div>
+        <div className="flex flex-row justify-end gap-2">
+          <ActionButton
+            size="sm"
+            type="button"
+            appearance="outlined"
+            onClick={onCancel}
+            disabled={submission.isSubmitting}
+          >
+            Cancel
+          </ActionButton>
+          <ActionButton
+            size="sm"
+            type="submit"
+            appearance="primary"
+            disabled={submission.isSubmitting}
+          >
+            {!submission.isSubmitting
+              ? !!member
+                ? 'Update Member'
+                : 'Add Member'
+              : !!member
+              ? 'Updating Member...'
+              : 'Adding Member...'}
```

Comment:
This one is a bit tricky to read through for me!

Reply:
yeah went a bit too crazy with the ternaries lol. refactored it into its own helper